blaze/server/serialization.py

```diff
@@ -34,6 +96,101 @@ def _coerce_str(bytes_or_str):
 )
 
 
+try:
+    import blosc
+    del blosc
+    compress = 'blosc'
+except ImportError:
+    compress = None
+
+
+def fastmsgpack_object_hook(ob):
+    typ = ob.get('typ')
+    if typ is None:
+        return ob
+    if typ == 'nat':
+        return pd.NaT
+    return pd_msgpack_object_hook(ob)
+
+
+if PY2:
+    def _l1(bs):
+        if isinstance(bs, unicode):
+            return bs.encode('latin1')
+        return bs
+else:
+    def _l1(bs):
+        return bs
+
+
+def fastmsgpack_data_loads(data):
+    raw = list(msgpack_unpack(
+        BytesIO(_l1(data)),
+        object_hook=fastmsgpack_object_hook,
+    ))
+    if len(raw) == 1:
+        return raw[0]
```

Comment:
All objects are sent as lists. This case is so that if I have only sent back 1 dataframe we will get the single dataframe instead of a list of length 1 containing the dataframe. This logic is borrowed from pandas.io.packers.read_msgpack. We are reimplementing this instead of using that function so that we can forward the object hook in pandas 0.16.